src/screens/NetworkSettings/CustomNetworkSettingsScreen.js

```diff
@@ -0,0 +1,260 @@
+import React, { useEffect, useState } from 'react';
+import { View, Text, StyleSheet, Image } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { isEmpty } from 'lodash';
+import { networkSettingsUpdate, networkSettingsUpdateErrors, networkSettingsUpdateReady } from '../../actions';
+import FeedbackModal from '../../components/FeedbackModal';
+import HathorHeader from '../../components/HathorHeader';
+import NewHathorButton from '../../components/NewHathorButton';
+import SimpleInput from '../../components/SimpleInput';
+import { NETWORKSETTINGS_STATUS } from '../../constants';
+import errorIcon from '../../assets/images/icErrorBig.png';
+import Spinner from '../../components/Spinner';
+
+const customNetworkSettingsTitleText = t`Custom Network Settings`.toUpperCase();
+const warningText = t`Any token outside mainnet network bear no value. Only change if you know what you are doing.`;
+const feedbackLoadingText = t`Updating custom network settings...`;
+const feedbackFailedText = t`There was an error while customizing network settings. Please try again later.`;
+
+/**
+ * Check if the network settings status is failed.
+ * @param {object} networkSettingsStatus - status from redux store
+ * @returns {boolean} - true if the status is failed, false otherwise
+ */
+const hasFailed = (networkSettingsStatus) => {
+  return networkSettingsStatus === NETWORKSETTINGS_STATUS.FAILED;
+};
+
+/**
+ * Check if the network settings status is loading.
+ * @param {object} networkSettingsStatus - status from redux store
+ * @returns {boolean} - true if the status is loading, false otherwise
+ */
+const isLoading = (networkSettingsStatus) => {
+  return networkSettingsStatus === NETWORKSETTINGS_STATUS.LOADING;
+};
+
+/**
+ * Verifies if the errorModel of the form has an error message. 
+ */
+function hasError(errorModel) {
+  return Object
+    .values({ ...errorModel })
+    .reduce((hasError, currValue) => hasError || !isEmpty(currValue), false);
```

Comment:
Can't you do something similar to this?

```
return Object.values(errorModel).length > 0
```

It's not wrong what you did but it seems more complicated.

Reply:
Oh! No. Sometimes the `errorModel` has attributes, however empty. This way we get both states: empty and deep empty.